### DISCUSSION TITLE
cms-wordpress

### DISCUSSION THREAD

**Discussion by Uranbold** [vercel/next.js#12767] [Help]
In today's release, there are lots of cms example added.

Please consider about the Official WordPress Next JS Example :)

**Comment by timneutkens** [comment]
Already in the works!

**Reply by Uranbold** [reply]
I can help if you want it. I know WP a little bit :) I was a Themeforest Premium theme author :)

**Reply by janus-reith** [reply]
@timneutkens Interesting, is there anything public to take a look a yet? I'm currently implementing that aswell, and had to take some workarounds to get preview pages working (Basically some kind of superuser that uses basic auth to connect to WP, as I found no proper way to use the wordpress session cross-site on the nextjs server when doing fallback preview generation for drafts).

Also, after sticking with the REST API initially, I switched to WPGraphQL Gutenberg which comes in handy to have access to the defined props in the editor, although it might also work to just parse them out of the raw data in the html comments that Gutenberg uses to store things.

In nextjs I then parse block by block, and decide wether to just render the inner html or do something special with that block type.
Although I never was much of a fan of Wordpress, I have high hopes for Gutenberg as an Editor and I believe there are some pretty huge things that can be achieved this way which seemed out of scope before.
E.g. I can have that simple drag and drop page building in Gutenberg even for more complex layouts, and enrich that with data coming from inside my React Context in Nextjs.
Just for example, someone could add a featured product to some landing page in Gutenberg, and instead of just having things like title and price static from that point on, I could evaluate that information when revalidating inside nextjs and enrich it with up to date product data fetched during getStaticProps.

**Reply by Uranbold** [reply]
@janus-reith You can try this -> https://github.com/postlight/headless-wp-starter

This is really great starter and they did the 2 different approaches with Rest API & GraphQL, Also they implemented it into docker :)

The code may outdated but really useful for starting and understanding.

I am also a huge fan of WordPress and Gutenberg seems cool. But the UX is really bad :P

**Reply by janus-reith** [reply]
@Uranbold Thanks, I also came across this some weeks ago but already have a basic integration implemented myself that takes care of more details, while that boilderplate doesn't do much of the specific things I was looking for. The preview route there circumvents issues by only executing the fetch for data on a client side mount, something I want to avoid. Apart from that I believe the way this sets up WP with the plugins might be handy, but provided no new insights for me.

**Comment by janus-reith** [comment]
For reference: https://github.com/zeit/next.js/pull/13194

**Comment by karandeepvirk85** [comment]
I am integrating Wordpress Backend with React theme. 
I really love wordpress Back-end admin panel and I am using WP-REST API. 
Can we have option to use word-press without theme as an API in modern frameworks like Angular and React.
Or Do I have to create theme function.php to add stuff like **_add_theme_support('post-thumbnails');_**
I am not sure if it is already there?
